generate/genfile/genfile_test.go

```diff
@@ -0,0 +1,526 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package genfile_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/madlambda/spells/assert"
+	"github.com/mineiros-io/terramate/errors"
+	"github.com/mineiros-io/terramate/generate/genfile"
+	"github.com/mineiros-io/terramate/hcl"
+	"github.com/mineiros-io/terramate/test"
+	errtest "github.com/mineiros-io/terramate/test/errors"
+	"github.com/mineiros-io/terramate/test/hclwrite"
+	"github.com/mineiros-io/terramate/test/sandbox"
+	"github.com/rs/zerolog"
+)
+
+func TestLoadGenerateFiles(t *testing.T) {
+	type (
+		hclconfig struct {
+			path string
+			add  fmt.Stringer
+		}
+		genFile struct {
+			body   string
+			origin string
+		}
+		result struct {
+			name string
+			file genFile
+		}
+		testcase struct {
+			name    string
+			stack   string
+			configs []hclconfig
+			want    []result
+			wantErr error
+		}
+	)
+
+	hcldoc := func(builders ...hclwrite.BlockBuilder) *hclwrite.Block {
+		return hclwrite.BuildHCL(builders...)
+	}
+	generateFile := func(builders ...hclwrite.BlockBuilder) *hclwrite.Block {
+		return hclwrite.BuildBlock("generate_file", builders...)
+	}
+	labels := func(labels ...string) hclwrite.BlockBuilder {
+		return hclwrite.Labels(labels...)
+	}
+	globals := func(builders ...hclwrite.BlockBuilder) *hclwrite.Block {
+		return hclwrite.BuildBlock("globals", builders...)
+	}
+	expr := func(name string, expr string) hclwrite.BlockBuilder {
+		return hclwrite.Expression(name, expr)
+	}
+	str := func(name string, val string) hclwrite.BlockBuilder {
+		return hclwrite.String(name, val)
+	}
```

Comment:
```suggestion
	hcldoc := hclwrite.BuildHCL
	labels := hclwrite.Labels
	expr := hclwrite.Expression
	str := hclwrite.String
	generateFile := func(builders ...hclwrite.BlockBuilder) *hclwrite.Block {
		return hclwrite.BuildBlock("generate_file", builders...)
	}
	globals := func(builders ...hclwrite.BlockBuilder) *hclwrite.Block {
		return hclwrite.BuildBlock("globals", builders...)
	}
```

just a simplification.